src/foam/core/Validation.js

```diff
@@ -45,8 +49,10 @@ foam.CLASS({
       class: 'Function',
       // TODO: it isn't normal for JS functions to have a 'js' prefix
       name: 'jsErr',
-      expression: function(errorString) {
-        return function() { return errorString; };
+      expression: function(errorString, errorMessage) {
+        return function(obj) { 
+          return errorMessage && obj ? obj[errorMessage] : errorString;
```

Comment:
Let's add a condition check on obj[errorMessage] and throw a warning if no string/message is found on the object.